File: 10k app/frontend/lib/data/models/nutrition_goal.dart
class NutritionGoal {
  final double goalCalories;
  final double goalProtein;
  final double goalCarbs;
  final double goalFat;

  const NutritionGoal({
    required this.goalCalories,
    required this.goalProtein,
    required this.goalCarbs,
    required this.goalFat,
  });

  factory NutritionGoal.fromJson(Map<String, dynamic> json) => NutritionGoal(
        goalCalories: (json['goalCalories'] as num).toDouble(),
        goalProtein: (json['goalProtein'] as num).toDouble(),
        goalCarbs: (json['goalCarbs'] as num).toDouble(),
        goalFat: (json['goalFat'] as num).toDouble(),
      );
}

File: 10k app/backend/src/main/java/com/gymflow/model/NutritionGoal.java
package com.gymflow.model;

import jakarta.persistence.*;
import lombok.*;

@Entity
@Table(name = "nutrition_goal")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class NutritionGoal {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private double goalCalories;

    @Column(nullable = false)
    private double goalProtein;

    @Column(nullable = false)
    private double goalCarbs;

    @Column(nullable = false)
    private double goalFat;
}
